server/src/main/java/org/elasticsearch/common/recycler/DequeRecycler.java

```diff
@@ -45,6 +45,12 @@ public DequeRecycler(C<T> c, Deque<T> queue, int maxSize) {
         return new DV(v, true);
     }
 
+    @Override
+    public int recycledCount() {
+        // TODO: Does not reflect that the currently allocated number of pages might be less.
```

Comment:
why not add tracking for this in `DequeRecycler`? Is this difficult to do?

Reply:
It is just work and I wanted to avoid making many changes before getting feedback (also I think it needs to be synchronized for the non-concurrent Deques).